File: ejercicios hechos/Python/ejercicios repeticiones/act5.py
#Programa: act5.py
#Propósito: Escribe un programa que pida el limite inferior y superior de un intervalo. Si el límite inferior es mayor que el superior lo tiene que volver a pedir. 
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 29/10/2019
#
#Variables a usar:
# lim_inf, lim_sup son los valores de los límites
# n1 será el número que introducimos
# sum_int es la suma del intervalo
# cont_int será el conteo
# igual_lim es el indicativo que es igual a un intervalo
#
#Algoritmo:
# Pedimos un intervalo
# El límite inferior debe ser menor que el superior
# Se piden numeros hasta que se introduce el  0
# Si el número es del intervalo se suma
# Si el numero no pertenece al intervalo, se cuenta
# Si el nímero es igual al límite, se índica
# Se muestra la suma de los numeros que pertenecen al intervalo
# Se muestra el contador de números que no pertenecen al intervalo
# Indicamos si se ha introducido un número igual a algún límite

print("Programa para operar con limite inferior y superior")
print("----------------------------------------------------\n")

#Leemos los datos

lim_sup = int(input("Introduce el límite superior: "))
lim_inf = int(input("Introduce el límite inferior: "))
sum_int = 0
cont_int = 0
igual_lim = False

# Comprobamos que no sea el inferior mayor al superior
while lim_inf > lim_sup:
    if lim_inf > lim_sup:
        print("El límite inferior no puede ser mayor al límite superior")
        lim_sup = int(input("Introduce el límite superior: "))
        lim_inf = int(input("Introduce el límite inferior: "))

n1 = int(input("Introduce un número para salir(0 para finalizar): "))

while n1 != 0:
    if n1 > lim_inf and n1 < lim_sup:
        sum_int = sum_int + n1
    else:
        cont_int = cont_int + 1
    
    if n1 == lim_inf or n1 == lim_sup:
        igual_lim = True
        
    n1 = int(input("Introduce un número para salir(0 para finalizar): "))

print(f"La suma de los números es: {sum_int}")
print(f"Los números fuera del intervalo son: {cont_int}")

if igual_lim:
    print("Algún número es igual a los límites del intervalo")
else:
    print("No hay ningún número igual a los límites del intervalo")

File: ejercicios hechos/Java/estructuras repetitivas/act5.java
/*Programa: act5.java
 * 
 *Propósito: Escribe un programa que pida el limite inferior y superior de 
 * un intervalo. Si el límite inferior es mayor que el superior lo tiene que 
 * volver a pedir. 

A continuación se van introduciendo números hasta que introduzcamos el 0. 
* Cuando termine el programa dará las siguientes informaciones:

La suma de los números que están dentro del intervalo (intervalo abierto).
Cuantos números están fuera del intervalo.
Informa si hemos introducido algún número igual a los límites del intervalo.
 * 
 *Autor: Jose Manuel Serrano Palomo.
 * 
 *Fecha: 10/11/2019
 *
 *Análisis:
 * Debemos clasificar unos intervalos en cuanto a si lo que introduce el usuario
 * esta dentro del intervalo, fuera o es uno de  los limites
 *
 *Diseño:
 * 1. Inicializo el acumulador, el contador y el indicado
 * 2. Repetir -> ller limInf y limSup hasta que limInf<limSup
 * 3. Leer num
 * 4. Mientras num sea distinto de 0
 * 5. Si num pertenece al intervalo -> acumula la suma
 * 6. Sino -> incremento contador
 * 7. Si es igual a algún intervalo -> Lo indico (indicador a Verdadero)
 * 8. Leer num
 * 9. Muestro la suma de los números pertenecientes al intervalo
 * 10. Muestro el contador de números no pertenecientes al intervalo
 * 11. Si el indicador = Verdadero -> Muestro "Un número = a intervalo"
 * 12. SiNo -> Muestro "No has introducido un número igual al intervalo
*/
import java.util.Scanner;
public class act5 {
    public static void main(String[] args) {
        System.out.println("Programa que define intervalos");
        System.out.println("------------------------------");
        Scanner s = new Scanner(System.in);
        int contFueraIntervalo = 0;
        boolean igualLimites = false;
        int limInf;
        int limSup;
        int num;
        int sumaDentroIntervalo = 0;

        do {
          System.out.print("Introduce el límite inferior del intervalo: ");
          limInf = s.nextInt();
          System.out.print("Introduce el límite superior del intervalo: ");
          limSup = s.nextInt();
          if (limInf>limSup) {
            System.out.println("ERROR: El límite inferior debe ser menor que el superior.");
          }
        } while (limInf>limSup);

        System.out.print("Introduce un número (0, para salir): ");
        num = s.nextInt();
        while (num!=0) {
          if (num>limInf && num<limSup) {
            sumaDentroIntervalo += num;
          } else {
            contFueraIntervalo++;
          }
          if (num==limInf || num==limSup) {
            igualLimites = true;
          }
          System.out.print("Introduce un número (0, para salir): ");
          num = s.nextInt();
        }

        System.out.println("La suma de los número dentro del intervalo es " + sumaDentroIntervalo);
        System.out.println("La cantidad de números fuera del intervalo es " + contFueraIntervalo);
        if (igualLimites) {
          System.out.println("Se ha introducido algún número igual a los límites del intervalo.");
        } else {
          System.out.println("No se ha introducido ningún número igual a los límites del intervalo.");
        }
    }
}
